Solaris cp doesn't know -a, but GNU does.
When I set PATH to reach /usr/gnu/bin/cp rust fails to build.

Not for upstream.
--- a/rust/Makefile.am
+++ b/rust/Makefile.am
@@ -72,11 +72,11 @@ $(CARGO_ENV) \
 		$(CARGO) build $(RELEASE) $(NIGHTLY_ARGS) \
 			--features "$(RUST_FEATURES)" $(RUST_TARGET)
 	if test -e $(RUST_SURICATA_LIBDIR)/suricata.lib; then \
-		cp -a $(RUST_SURICATA_LIBDIR)/suricata.lib \
+		gcp -a $(RUST_SURICATA_LIBDIR)/suricata.lib \
 			$(RUST_SURICATA_LIBDIR)/${RUST_SURICATA_LIBNAME}; \
 	fi
 	if test -e $(RUST_SURICATA_LIBDIR)/libsuricata.a; then \
-		cp -a $(RUST_SURICATA_LIBDIR)/libsuricata.a \
+		gcp -a $(RUST_SURICATA_LIBDIR)/libsuricata.a \
 			$(RUST_SURICATA_LIBDIR)/${RUST_SURICATA_LIBNAME}; \
 	fi
 	$(MAKE) gen/rust-bindings.h
